gammapy/makers/background/reflected.py

```diff
@@ -140,134 +157,112 @@ def region_angular_size(self):
-
-    def reset_cache(self):
-        """Reset cached properties"""
-        for name, value in self.__class__.__dict__.items():
-            if isinstance(value, lazyproperty):
-                self.__dict__.pop(name, None)
-
-    def run(self):
+    def run(self, region, center):
         """Find reflected regions.
 
+        Parameters
+        ----------
+        region : `~regions.SkyRegion`
+            Region to rotate
+        center : `~astropy.coordinates.SkyCoord`
+            Rotation point
+
         Returns
         -------
         regions : list of `SkyRegion`
             Reflected regions
+        wcs: `~astropy.wcs.WCS`
+            WCS for the determined regions
         """
-        self.reset_cache()
         regions = []
 
-        angle = self.angle_min + self.min_distance_input
+        reference_geom = self._create_reference_geometry(region, center)
+        center_pixel = self._get_center_pixel(center, reference_geom)
+        angle_min, angle_max = self._get_angle_range(
+            region=region, reference_geom=reference_geom, center_pix=center_pixel,
+        )
+
+        region_pix = self._get_region_pixels(region, reference_geom)
+        excluded_pixels = self._get_excluded_pixels(reference_geom)
 
-        while angle < self.angle_max:
-            region_test = self.region_pix.rotate(self.center_pix, angle)
+        angle = angle_min + self.min_distance_input
+        while angle < angle_max:
+            region_test = region_pix.rotate(center_pixel, angle)
 
-            if not np.any(region_test.contains(self.excluded_pix_coords)):
-                region = region_test.to_sky(self.geom_ref.wcs)
+            if not np.any(region_test.contains(excluded_pixels)):
+                region = region_test.to_sky(reference_geom.wcs)
                 regions.append(region)
 
                 if len(regions) >= self.max_region_number:
                     break
 
-                angle += self.angle_min
+                angle += angle_min
             else:
                 angle += self.angle_increment
 
-        return regions
+        return regions, reference_geom.wcs
 
 
 class ReflectedRegionsBackgroundMaker(Maker):
     """Reflected regions background maker.
 
     Parameters
     ----------
-    angle_increment : `~astropy.coordinates.Angle`, optional
-        Rotation angle applied when a region falls in an excluded region.
-    min_distance : `~astropy.coordinates.Angle`, optional
-        Minimal distance between two consecutive reflected regions
-    min_distance_input : `~astropy.coordinates.Angle`, optional
-        Minimal distance from input region
-    max_region_number : int, optional
-        Maximum number of regions to use
-    exclusion_mask : `~gammapy.maps.WcsNDMap`, optional
-        Exclusion mask
-    binsz : `~astropy.coordinates.Angle`
-        Bin size of the reference map used for region finding.
+    regions_finder: RegionsFinder
```

Comment:
Maybe specify list of possible choices.
Also specify default choice.

Reply:
The `kwargs` argument should be listed in the parameters list. 
And possibly an explicit link to the `ReflectedRegionsFinder` if it is the default.

Reply:
Shouldn't the `mask` be an argument of the `Maker` itself since it will be an argument of all `RegionsFinder`?

Reply:
Yes, that's a good point. Probably the `exclusion_mask` should also be moved to the `run` method.

Reply:
Done. How that cleaned up the test was a good sign that again, this is the preferable API